moveit_ros/occupancy_map_monitor/src/occupancy_map_monitor.cpp

```diff
@@ -39,124 +39,90 @@
 #include <moveit_msgs/srv/load_map.hpp>
 #include <moveit/occupancy_map_monitor/occupancy_map.h>
 #include <moveit/occupancy_map_monitor/occupancy_map_monitor.h>
+#include <moveit/occupancy_map_monitor/occupancy_map_monitor_middleware_handle.hpp>
 #include <boost/bind.hpp>
 
 namespace occupancy_map_monitor
 {
+namespace
+{
 static const rclcpp::Logger LOGGER = rclcpp::get_logger("moveit.ros.occupancy_map_monitor");
+}
 
 OccupancyMapMonitor::OccupancyMapMonitor(const rclcpp::Node::SharedPtr& node, double map_resolution)
-  : map_resolution_(map_resolution), debug_info_(false), mesh_handle_count_(0), node_(node), active_(false)
+  : OccupancyMapMonitor(std::make_unique<OccupancyMapMonitorMiddlewareHandle>(node), nullptr, "", map_resolution)
 {
-  initialize();
 }
 
 OccupancyMapMonitor::OccupancyMapMonitor(const rclcpp::Node::SharedPtr& node,
                                          const std::shared_ptr<tf2_ros::Buffer>& tf_buffer,
                                          const std::string& map_frame, double map_resolution)
-  : tf_buffer_(tf_buffer)
+  : OccupancyMapMonitor(std::make_unique<OccupancyMapMonitorMiddlewareHandle>(node), tf_buffer, map_frame,
+                        map_resolution)
+{
+}
+
+OccupancyMapMonitor::OccupancyMapMonitor(std::unique_ptr<MiddlewareHandle> middleware_handle,
+                                         const std::shared_ptr<tf2_ros::Buffer>& tf_buffer,
+                                         const std::string& map_frame, double map_resolution)
+  : middleware_handle_(std::move(middleware_handle))
+  , tf_buffer_(tf_buffer)
   , map_frame_(map_frame)
   , map_resolution_(map_resolution)
   , debug_info_(false)
   , mesh_handle_count_(0)
-  , node_(node)
-{
-  initialize();
-}
-
-void OccupancyMapMonitor::initialize()
+  , active_(false)
 {
-  /* load params from param server */
-  if (map_resolution_ <= std::numeric_limits<double>::epsilon())
```

Comment:
I missed this.  The default value is 0.0 which causes it to look for the parameter.